### PR TITLE
Add a section on ReferenceEvaluator in the tutorial

### PR DESCRIPTION
**Pull request by xadupre**
### Description
Extend the documentation.

### Motivation and Context
Better documentation.

### CODE REVIEW

File: docs/docsgen/source/intro/concepts.rst [hunk 1]
```diff
@@ -31,6 +31,9 @@ to deploy a machine learning model in production. An ONNX interpreter
 in the environment where it is deployed. With ONNX, it is possible
 to build a unique process to deploy a model in production and independent
 from the learning framework used to build the model.
+*onnx* implements a python runtime (see :ref:`l-reference-implementation`)
+to help evaluating ONNX models
```
Comment (gramalingam) [inline docs/docsgen/source/intro/concepts.rst:35]: nit: change to
```
a python runtime that can be used to evaluate ONNX models and to evaluate ONNX ops.
This is intended to clarify the semantics of ONNX and to help understand/debug ONNX tools
and converters. It is not intended to be used for production and efficiency/performance is
not a goal.
```

File: docs/docsgen/source/intro/python.rst [hunk 2]
```diff
@@ -1179,6 +1180,279 @@ a Reshape operator. Shape inference only works if the shape is constant.
 If not constant, the shape cannot be easily inferred unless
 the following nodes expect specific shape.
 
+Evaluation and Runtime
+======================
+
+An ONNX model can be used to move a model from one library to another
```
Comment (gramalingam) [inline docs/docsgen/source/intro/python.rst:1186]: How about
```
The ONNX standard allows frameworks to export trained models in ONNX format,
and enables inference using any backend that supports the ONNX format.
`onnxruntime` is one efficient option ...
```

File: docs/docsgen/source/intro/python.rst [hunk 3]
```diff
@@ -1179,6 +1180,279 @@ a Reshape operator. Shape inference only works if the shape is constant.
...
+or it can be directly evaluated with an ONNX runtime.
+:epkg:`onnxruntime` is one efficient option. It is available in many platforms.
+It is optimized for fast inference. Its coverage can be tracked on
+`ONNX Backend Dashboard <http://onnx.ai/backend-scoreboard/>`_.
+*onnx* implements a python runtime
+useful to help understand a model.
```
Comment (gramalingam) [inline docs/docsgen/source/intro/python.rst:1192]: Add `but is not intended for production use`.

File: docs/docsgen/source/intro/python.rst [hunk 4]
```diff
@@ -1179,6 +1180,279 @@ a Reshape operator. Shape inference only works if the shape is constant.
...
 the following nodes expect specific shape.
 
+Evaluation and Runtime
+======================
+
+An ONNX model can be used to move a model from one library to another
+or it can be directly evaluated with an ONNX runtime.
+:epkg:`onnxruntime` is one efficient option. It is available in many platforms.
+It is optimized for fast inference. Its coverage can be tracked on
+`ONNX Backend Dashboard <http://onnx.ai/backend-scoreboard/>`_.
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/docsgen/source/intro/python.rst:1190]: done

File: docs/docsgen/source/intro/concepts.rst (hunk 1 above)
Comment (xadupre) [reply docs/docsgen/source/intro/concepts.rst:35]: done

File: docs/docsgen/source/intro/python.rst (hunk 2 above)
Comment (xadupre) [reply docs/docsgen/source/intro/python.rst:1186]: done

File: docs/docsgen/source/intro/python.rst (hunk 3 above)
Comment (xadupre) [reply docs/docsgen/source/intro/python.rst:1192]: done

**Review by gramalingam — APPROVED** [review]

**Pull request merged**